### PR TITLE
Deprecate slicedim(…) to copy(selectdim(…)); always return a view

### LINKED ISSUE #18326: slicedim and mapslices arguments inconsistent
In slicedim() you tell which dimension equals a value:

```
a=rand(6,5)
slicedim(a, 1, n) # returns n row a[n, :]
slicedim(a, 2, n) # returns n column a[:, n]
```

On the other hand in mapslices you tell where the dots (:) are, so:
`mapslice(f, a, 1) # a[:, n]`
Would pass columns to function f.

### PR DESCRIPTION
**Pull request by mbauman**
Fixes #18326.

Edit: this now couples a behavior change in with the name change.  The deprecation gives us a convenient time to improve its behavior to use views instead of getindex.

### CODE REVIEW

**Review by ararslan — DISMISSED** [review]

**Comment by stevengj** [comment]
I didn't even know this function existed; we don't use it anywhere in Base, but it seems to be used in a dozen or so packages.

(I wonder if there should be a `SubArray` version of this?)

**Comment by mbauman** [comment]
Yeah, see my comments at https://github.com/JuliaLang/julia/issues/18326#issuecomment-329907404 — one package defines its own `slicedim` but uses views instead.  Arguably that's the much more valuable behavior.

This function is so old I'd bet it predates views.

File: base/deprecated.jl [hunk 1]
```diff
@@ -1360,6 +1360,9 @@ end
 # issue #25928
 @deprecate wait(t::Task) fetch(t)
 
+# PR #26008
+@deprecate slicedim(A::AbstractArray, d::Integer, i) selectdim(A, d, i)
```
Comment (ararslan) [inline base/deprecated.jl]: If it's a straight renaming of one to the other you can just use `@deprecate slicedim selectdim`

Comment (mbauman) [reply base/deprecated.jl]: Yeah, I suppose it's effectively the same — that would define and export a broader signature, but it'd still hit a method error later on.

**Comment by mbauman** [comment]
You know what, I think views are indeed the better functionality here… and a name change is a great time to do that.  I pushed a new commit for that.

File: base/deprecated.jl [hunk 2]
```diff
@@ -26,7 +26,7 @@ macro deprecate(old, new, ex=true)
         newname = Expr(:quote, new)
         Expr(:toplevel,
             ex ? Expr(:export, esc(old)) : nothing,
-            :(function $(esc(old))(args...)
+            :(@__doc__ function $(esc(old))(args...)
```
Comment (mbauman) [inline base/deprecated.jl]: Watch out, I snuck in the ability to document deprecated functions :)

Comment (mbauman) [reply base/deprecated.jl]: Oh, never mind, it bails on bootstrapping and I don't feel strongly enough to battle that one out.

File: NEWS.md [hunk 3]
```diff
@@ -646,6 +646,10 @@ Deprecated or removed
   * Using Bool values directly as indices is now deprecated and will be an error in the future. Convert
     them to `Int` before indexing if you intend to access index `1` for `true` and `0` for `false`.
 
+  * `slicedim(A, d, i)` has been deprecated in favor of `copy(selectdim(A, d, i)`. The new
```
Comment (nalimilan) [inline NEWS.md]: Missing closing parenthesis.

**Comment by StefanKarpinski** [comment]
Given that it always returns a view now, maybe it should be called `dimview` :trollface:

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**